### PR TITLE
feat: refactor transactions list

### PR DESCRIPTION
**Pull request by He1DAr**
Refactor transactions list
Remove `Transaction | MempoolTransaction` abstraction

### CODE REVIEW

File: src/features/transactions-filter/hooks/use-filter-scope.ts [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+
 export const useFilterScope = () => {
   const router = useRouter();
-  const filterScope =
-    router.pathname === '/transactions'
-      ? TxFilterTypes.TransactionsPageTxFilter
-      : router.pathname === '/address/[principal]'
-      ? TxFilterTypes.AddressTxFilter
-      : TxFilterTypes.HomepageTxFilter;
+  console.log('router', router);
```
Comment (beguene) [inline src/features/transactions-filter/hooks/use-filter-scope.ts:16]: to be removed

File: src/modules/TransactionList/components/TxsList.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  );
+});
+
+interface MempoolTxsListProps extends TxsListCommonProps {
+  response: UseInfiniteQueryResult<MempoolTransactionsListResponse>;
+}
+
+export const MempoolTxsList: FC<MempoolTxsListProps> = memo(
+  ({ response, showFooter, infinite, limit }) => {
+    console.log('[debug] rendering MempoolTxsList');
```
Comment (beguene) [inline src/modules/TransactionList/components/TxsList.tsx:71]: to be removed (unless you need it for prod)

File: src/modules/TransactionList/components/TxsListWithTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import { MempoolTxsList, TxsList } from './TxsList';
+import { useTransactionList } from '../hooks/useTransactionList';
+
+const TX_TABS = 'tabs/tx-list';
+
+export const TxsListWithTabs: React.FC<{
+  limit: number;
+  infinite?: boolean;
+}> = memo(({ limit, infinite }) => {
+  console.log('[debug] rendering TxsListWithTabs');
```
Comment (beguene) [inline src/modules/TransactionList/components/TxsListWithTabs.tsx:18]: to remove (unless you need it for prod debug)

**Review by beguene — APPROVED** [review]
Nicely done @He1DAr ! Love the specific component just for transactions and the general code cleanup. 🧹 
I tested the pages, works fine so far.

File: src/modules/TransactionList/components/TxsList.tsx (hunk 2 above)
Comment (He1DAr) [reply src/modules/TransactionList/components/TxsList.tsx:71]: I'll remove these ones shortly

**Pull request merged**